File: sdk/prady-sdk/src/notify.ts
async function readJson<T>(response: Response): Promise<T> {
  if (!response.ok) {
    const detail = await response.text();
    throw new Error(detail || `HTTP ${response.status}`);
  }
  return (await response.json()) as T;
}

export class KryosNotify {
  constructor(private readonly apiBase: string = "http://localhost:8007") {}

  async send(title: string, body: string, severity: "info" | "warning" | "critical" = "info"): Promise<void> {
    const response = await fetch(`${this.apiBase}/notify`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ title, body, severity, source: "sdk-app" }),
    });
    await readJson<{ ok: boolean }>(response);
  }
}

File: sdk/prady-sdk-python/kryos_sdk/notify.py
from __future__ import annotations

import httpx


class KryosNotify:
    def __init__(self, api_base: str = "http://localhost:8007") -> None:
        self.api_base = api_base

    async def send(self, title: str, body: str, severity: str = "info") -> None:
        async with httpx.AsyncClient() as client:
            resp = await client.post(
                f"{self.api_base}/notify",
                json={"title": title, "body": body, "severity": severity, "source": "sdk-app"},
            )
        resp.raise_for_status()
